pkg/etcd/etcd.go

```diff
@@ -1182,26 +1201,36 @@ func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client corev1.CoreV1Interfa
 		}
 	}
 
-	updatedNode := *node
-	if find, condition := nodeUtil.GetNodeCondition(&updatedNode.Status, etcdStatusType); find >= 0 {
-		if condition.Status == newCondition.Status {
-			logrus.Debugf("Member %s is not changing etcd status condition", memberName)
+	if !isMember {
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "False",
+			Reason:  "NotAMember",
+			Message: "Node is not a member of the etcd cluster",
+		}
+	}
+
+	if find, condition := nodeUtil.GetNodeCondition(&node.Status, etcdStatusType); find >= 0 {
+		if condition.Status == newCondition.Status && isMember {
+			logrus.Debugf("Node %s is not changing etcd status condition", memberName)
 			condition.LastHeartbeatTime = metav1.Now()
-			return nodeHelper.PatchNodeStatus(client, types.NodeName(node.Name), node, &updatedNode)
+			err := nodeHelper.SetNodeCondition(client, types.NodeName(node.Name), *condition)
+			return err
```

Comment:
This helper has the same return signature as this function now, so there's no need to temporarily assign to a var.
```suggestion
			return nodeHelper.SetNodeCondition(client, types.NodeName(node.Name), *condition)
```